pkg/stomp/StompContext.php

```diff
@@ -181,6 +190,10 @@ public function createConsumer(Destination $destination): Consumer
      */
     public function createProducer(): Producer
     {
+        if ($this->transient && true == $this->stomp) {
+            $this->stomp->disconnect();
+        }
```

Comment:
Simply triggering a disconnect is enough, as underlying layers can rely on their `false` state around connections, things pick up without any additional fuss at the enqueue level.